src/sardana/macroserver/recorders/h5storage.py

```diff
@@ -325,6 +330,54 @@ def _endRecordList(self, recordlist):
 
         env = self.currentlist.getEnviron()
         nxentry = self.fd[self.entryname]
+
+        for dd in env['datadesc']:
+            dtype = dd.getDtype()
+            name = dd.toDict()['name']
+
+            # If h5file scheme is used: Creation of a Virtual Dataset
+            if dtype == "str":
+                measurement = nxentry['measurement']
+                first_reference = measurement[name][0]
+
+                if not first_reference.startswith("h5file://"):
+                    continue
+                if not VDS_available:
+                    msg = ("VDS not available in this version of h5py, "
+                           "{0} will be stored as string reference")
+                    msg.format(name)
+                    self.warning(msg)
+                    continue
+
+                bk_name = "_" + name
+                measurement[bk_name] = measurement[name]
+                nb_points = measurement[name].size
+                filename = first_reference.split("://")[1]
+                with h5py.File(filename, 'r') as f:
+                    (dim_1, dim_2) = f['dataset'].shape
+
+                layout = h5py.VirtualLayout(shape=(nb_points,
+                                                   dim_1, dim_2),
+                                            dtype='f')
+                for i in range(nb_points):
+                    reference = measurement[name][i]
+                    filename = reference.split("://")[1]
+                    vsource = h5py.VirtualSource(filename, "dataset",
```

Comment:
"dataset" name of the _dataset_ is just an example from the test. We need to extract the real dataset name from the URI, based on `::` split. Regular expressions may be useful.